llvm/include/llvm/IR/IntrinsicsAMDGPU.td

```diff
@@ -1939,6 +1939,27 @@ def int_amdgcn_s_buffer_prefetch_data : DefaultAttrsIntrinsic <
 
 } // defset AMDGPUBufferIntrinsics
 
+// A wrapper around raw_ptr_buffer_load_lds that takes the global offset
+// from the addrspace(7) pointer argument.
+def int_amdgcn_buffer_fat_ptr_load_lds : Intrinsic <
```

Comment:
Is this really the name you want? None of the other buffer ptr intrinsics have "fat" in the name.

Reply:
Maybe I'm misunderstanding what this is for. Is it fundamentally different from the existing int_amdgcn_raw_ptr_buffer_load_lds ?

Reply:
This is for `ptr addrspace(7)`, the "buffer fat pointer", which carries the offset in the pointer type 

It's a wrapper around raw_ptr_buffer_load_lds

Reply:
To me it seems like we might want to call the AS8 version something like `buffer.resource.load` and the AS7 `buffer.ptr.load`.

Reply:
Unfortunately, we've already got the `raw.ptr` and `struct.ptr` convention for AS8, so I don't want to cause confusion

Reply:
(I'm open to bikesheding about the name though)

Reply:
(On the other hand, we already use `buffer.rsrc` in `make.buffer.rsrc` ... yeah, there's room for weirdness there)

Reply:
Yeah when I did the opaque type I used `buffer_rsrc` because that was the name used in our guide.

Reply:
Yeah, no, makes sense

I don't think we can rename the existing intrinsic because of backcompat - or at least that'd be a separate PR - so ... do you have thoughts on an alternate name here?

Reply:
Still should not have fat in the name, and we can easily rename and auto upgrade intrinsics

Reply:
I'd expect these to just be mangled on the pointer type and use the existing intrinsic?

Reply:
I don't think this is used anywhere yet so I can revert it off tomorrow if we'd like

Reply:
Maybe not global... for cases like this in the past, we just had an unqualified intrinsic and only drove off the pointer type mangling. Just stick to buffer terminology, I suppose (and then just permit managing for 7/8). Intrinsics with unmangled pointer types are always trouble

Reply:
7 and 8 fake different arguments though